src/main/java/duke/command/UpdatePatientCommand.java

```diff
@@ -12,32 +12,41 @@
 
 public class UpdatePatientCommand extends Command {
 
-    private int Id;
-    private String targetInfo;
-    private String updatedValue;
-
-    public UpdatePatientCommand(int Id , String targetInfo , String updatedValue) {
-        super();
-        this.Id = Id;
-        this.targetInfo = targetInfo;
-        this.updatedValue = updatedValue;
-    }
+    private String command;
+
+    public UpdatePatientCommand(String command) { this.command = command; }
 
     @Override
     public void execute(PatientTaskList patientTask, TaskManager tasks, PatientManager patientManager, Ui ui, PatientTaskStorage patientTaskStorage, TaskStorage taskStorage, PatientStorage patientStorage) throws DukeException {
-        Patient targetPatient = patientManager.getPatient(Id);
-            if (targetInfo.equals("name")) {
-                targetPatient.setName(updatedValue);
-            } else if (targetInfo.equals("nric")) {
-                targetPatient.setNric(updatedValue);
-            } else if (targetInfo.equals("room")) {
-                targetPatient.setRoom(updatedValue);
-            } else {
-                throw new DukeException("No such Patient information existed. Please Enter a valid Patient information");
+        String[] tempCommand = command.split(" ", 3);
+        char firstChar = tempCommand[0].charAt(0);
+        if (firstChar == '#'){
```

Comment:
There is no else statement for this condition. Which means, for command `update patient 1 Name Huang`, nothing will be done

Reply:
I suggest to check for existence of '#' in `Parser.parseUpdatePatient()` if the UpdatePatientCommand only supports update by id for now. If the '#' is missing, throw an exception to remind user instead.

Reply:
Or you can support both 
1. `update patient #<id> <header> <value>`
2. `update patient <name> <header> <value>` (using the same approach as in deleteTask/deletePatient)

Reply:
Okay. For now I will just throw exception first. Have made the changes.